src/modules/esl-toggleable/test/toggleable.dispatcher.test.ts

```diff
@@ -0,0 +1,109 @@
+import {ESLToggleableDispatcher, ESLToggleable} from '../core';
+import {ESLEventUtils} from '../../esl-event-listener/core/api';
+
+describe('ESLToggleableDispatcher custom element', () => {
+  ESLToggleableDispatcher.init();
+  const $dispatcher = document.querySelector(ESLToggleableDispatcher.is) as ESLToggleableDispatcher;
+
+  const groupAttr1 = 'test-group1';
+  const groupAttr2 = 'test-group2';
+
+  const $el1 = ESLToggleable.create();
+  const $el2 = ESLToggleable.create();
+
+  $el1.setAttribute('group', groupAttr1);
+  $el2.setAttribute('group', groupAttr1);
+
+  document.body.appendChild($el1);
+  document.body.appendChild($el2);
+
+  beforeAll(() => {
+    jest.useFakeTimers();
+    ESLToggleable.register();
+  });
+
+  describe('ESLToggleableDispatcher instance', () => {
+    test('should initialize with correct tag name and default root as body', () => {
+      expect($dispatcher).toBeInstanceOf(ESLToggleableDispatcher);
+      expect($dispatcher.root).toBe(document.body);
+    });
+
+    test('shouldn`t initialize on the same level twice', () => {
+      ESLToggleableDispatcher.init();
+      expect(document.querySelectorAll(ESLToggleableDispatcher.is)).toHaveLength(1);
```

Comment:
Conflicts with a global state which makes this unit test case invalid by violation of isolation